### ISSUE TITLE
[Bug]: Compressed indexes always saved in default tablespace

### ISSUE THREAD

**Issue by kounoupaki** [timescale/timescaledb#4000]
### What type of bug is this?

Other

### What subsystems and features are affected?

Compression

### What happened?

When attaching multiple tablespaces to a hypertable, all uncompressed chunks and indexes are automatically saved in the selected tablespaces as expected. However, when we compress a chunk, the compressed chunk indexes are always saved in the default tablespace and not in one of the attached tablespaces of the hypertable.

For example, we created a hypertable with space dimension on device_id and 4 partitions:
```sql
 CREATE TABLE device_data (
	device_id int4 NOT NULL,
	value float8 NULL,
	"timestamp" timestamp NOT NULL,
	ingestion_timestamp timestamp NULL,
	value_string text NULL
);

SELECT create_hypertable('device_data', 
'timestamp', 'device_id', 4, chunk_time_interval => INTERVAL '6 hours');
```
We attached 4 tablespaces to our hypertable that correspond to 4 disks:
```sql
SELECT attach_tablespace('disk1', 'device_data');
SELECT attach_tablespace('disk2', 'device_data');
SELECT attach_tablespace('disk3', 'device_data');
SELECT attach_tablespace('disk4', 'device_data');

select * from show_tablespaces('device_data');
show_tablespaces|
----------------+
disk1           |
disk2           |
disk3           |
disk4           |
```

After filling some data, we checked that all uncompressed chunks and indexes are saved in the 4 disks as expected. For example,
```sql
select schemaname, tablename, tablespace from pg_tables;
schemaname           |tablename         |tablespace|
---------------------+------------------+----------+
_timescaledb_internal|_hyper_6_456_chunk|disk2     |

select schemaname, tablename, indexname, tablespace from pg_indexes;
schemaname           |tablename         |indexname                                                   |tablespace|
---------------------+------------------+------------------------------------------------------------+----------+
_timescaledb_internal|_hyper_6_456_chunk|_hyper_6_456_chunk_device_data_timestamp_idx             |disk2     |
_timescaledb_internal|_hyper_6_456_chunk|_hyper_6_456_chunk_device_data_device_id_timestamp_idx    |disk2     |
```

After enabling compression and compressing one chunk we noticed that while the compressed chunk is saved in one of the attached tablespaces, the compressed chunk index is saved in the default tablespace.
```sql
ALTER TABLE  device_data
  SET (timescaledb.compress,
       timescaledb.compress_segmentby = 'device_id');

SELECT compress_chunk('_timescaledb_internal._hyper_6_456_chunk');
```
Result of query in pg_tables:
```
schemaname           |tablename                  |tablespace|
---------------------+---------------------------+----------+
_timescaledb_internal|_hyper_6_456_chunk         |disk2     |
_timescaledb_internal|compress_hyper_7_1697_chunk|disk2     |
```
Result of query  in pg_indexes:
```
schemaname           |tablename                  |indexname                                                      |tablespace|
---------------------+---------------------------+---------------------------------------------------------------+----------+
_timescaledb_internal|_hyper_6_456_chunk         |_hyper_6_456_chunk_db_measurement_timestamp_idx                |disk2     |
_timescaledb_internal|_hyper_6_456_chunk         |_hyper_6_456_chunk_db_measurement_asset_id_timestamp_idx       |disk2     |
_timescaledb_internal|_hyper_6_456_chunk         |_hyper_6_456_chunk_db_measurement_qid_asset_id_timestamp_idx   |disk2     |
_timescaledb_internal|compress_hyper_7_1697_chunk|compress_hyper_7_1697_chunk__compressed_hypertable_7_asset_id__|          |
```

**Tested using:** 
- TimescaleDB Single Node HA setup for k8s using the provided [helm charts](https://github.com/timescale/timescaledb-kubernetes) with tag version pg14.1-ts2.5.1-p2

### TimescaleDB version affected

2.5.1

### PostgreSQL version used

14.1

### What operating system did you use?

Ubuntu 18.04

### What installation method did you use?

Docker

### What platform did you run on?

On prem/Self-hosted

**Comment by akuzm** [comment]
I can reproduce this, the index ends up in the default tablespace. For the record, here's the query I used to generate the data:
```
insert into device_data select random() * 100, ((random() * 100)::numeric % 77) * 0.37, now() + interval '1 month 3 day' * random(), now() + interval '7 weeks' * random(), random()::text from generate_series(1, 10000);
```

@gayyappan IIRC you know the most about the compression, is this something we should fix?

**Comment by gayyappan** [comment]
@akuzm Yes, this should be fixed.

**Comment by kounoupaki** [comment]
I don't know if it is somehow related to this or a completely different issue but I also noticed that when moving compressed chunks to other tablespaces using the move_chunk function, the indexes are not moved to the index_destination_tablespace. I tried this using a different hypertable with no attached tablespaces, so all chunks and indexes were initially saved in the default tablespace. 
After compressing the chunks, I tried to move them to a historical tablespace:
```sql
SELECT move_chunk(chunk => i, destination_tablespace => 'historical', index_destination_tablespace => 'historical') from show_chunks('device_data2', older_than => interval '2 years') i;
```
However, although the chunks (both compressed and the original) were moved to the historical tablespace, both the compressed table indexes (based in the segment by columns) and the non-compressed indexes that existed on the original table remained in the default tablespace.
